uswds/uswds
# USWDS - Card: Fix header first media overlap

# Summary

Fixed a margin bug on header-first variants that caused exdent card media to overlap headers.

## Breaking change

This is not a breaking change.

## Related issue

Closes #5401 

## Related pull requests

[Changelog →](https://github.com/uswds/uswds-site/pull/2227)

>**Note***
>Bug discovered while working on #5325 but has no overlap; They can be merged separately without error.

## Preview link

Preview link:
[Demo repo →](https://federalist-d5e7c07c-6ffa-4b8e-935f-49a7df24a505.sites.pages.cloud.gov/preview/uswds/uswds-sandbox/cm-card-header-first-media-test/)
[Card component →](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/cm-media-header-cutoff/iframe.html?args=&id=components-card--default)

## Problem statement

When header first variants for card components have large borders, the media overlaps the header. This is caused due to exdent styles setting a negative top margin to counteract the border. Because header-first variants will always have the media after the header, we should remove the negative top margin for this variant.

## Solution

Specify `0` top margin on exdent media within header-first variants.


## Testing and review

1. View [demo repo preview →](https://federalist-d5e7c07c-6ffa-4b8e-935f-49a7df24a505.sites.pages.cloud.gov/preview/uswds/uswds-sandbox/cm-card-header-first-media-test/)
2. Confirm media does not overlap header
3. Inspect `usa-card__media--exdent` class
4. Remove `top-margin: 0`.
5. Confirm media shifts to overlap header
6. _(optional)_ - Checkout test repo and update `$theme-card-border-width` in settings to various values and confirm header is always visible
   - [Demo repo →](https://github.com/uswds/uswds-sandbox/pull/107)

### Testing checklist 
- [ ] header-first card variants media never overlaps header
- [ ] Logic behind solution is valid: `Media on header-first cards should never have negative margin`

## Screenshots

<details>

<summary>Screenshots</summary>

### Before
![image](https://github.com/uswds/uswds/assets/25211650/c83bbb39-e333-48fd-8e08-74c3107d1c57)


### After
![image](https://github.com/uswds/uswds/assets/25211650/44a34298-aa19-499d-bd08-ade01909fdc6)

</details>
